mesonbuild/interpreter/interpreter.py

```diff
@@ -2845,10 +2847,12 @@ def check_clang_asan_lundef(self) -> None:
             return
         if (self.coredata.options[OptionKey('b_lundef')].value and
                 self.coredata.options[OptionKey('b_sanitize')].value != 'none'):
-            mlog.warning('''Trying to use {} sanitizer on Clang with b_lundef.
-This will probably not work.
-Try setting b_lundef to false instead.'''.format(self.coredata.options[OptionKey('b_sanitize')].value),
-                         location=self.current_node)
+            mlog.warning(textwrap.dedent(f'''\
+                    Trying to use
+                    {self.coredata.options[OptionKey('b_sanitize')].value }
+                    sanitizer on Clang with b_lundef.  This will probably not work.
+                    Try setting b_lundef to false instead.'''),
+                location=self.current_node, fatal=False)
```

Comment:
If it won't work, why isn't it a fatal warning to make you stop doing it? ISTR that "won't work" means linker errors, not "clang just doesn't set up sanitizer support".

Reply:
We've had reports of it working on some OSes and failing on others. I don't konw what the difference between them is, but I can drop this change